### PR TITLE
Prefer `nodenv shell` over package.json

### LINKED ISSUE #23: Let NODENV_VERSION take precedence
Like `.node-version`, it'd be useful if `NODENV_VERSION` could take precedence. I have a `package.json` which allows Node.js `>=4`. This plugin automatically selects the highest version I have installed, which makes it hard to test on older Node.js versions.

### PR DESCRIPTION
**Pull request by novemberborn**
Fixes #23. The relevant commit is 28bbfe594f44af45f1240ce0c88f03877bb611be, but I've also included some other cleanup and updates which I encountered during setup.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ Once you've installed the plugin you can verify that it's working by `cd`ing int
 
 ## Contributing
 
-To run tests, install [bats](https://github.com/sstephenson/bats) and [nodenv](https://github.com/nodenv/nodenv), then run `bats test`  in the base directory of this plugin
+To run tests, clone this repository and install dependencies using `npm`. Start a shell **in which `nodenv init -` has not been executed**, then run `npm test` in the base directory of this plugin.
```
Comment (hurrymaplelad) [inline README.md]: Any context on this bit? Tests still pass for me even after running `nodenv init -`

File: bin/nodenv-package-json-engine [hunk 2]
```diff
@@ -44,7 +44,7 @@ find_package_json_path() {
 }
 
 should_find_in_package_json() {
-  [ -z "$(nodenv local 2>/dev/null)" ]
+  [[ -z "$(nodenv local 2>/dev/null)" ]] && ([[ -z ${NODENV_VERSION+x} ]] || [[ "${NODENV_VERSION}" == "$(nodenv global 2>/dev/null)" ]])
```
Comment (hurrymaplelad) [inline bin/nodenv-package-json-engine]: This seems a bit brittle. I can't explicitly set `nodenv shell` to the same value as `nodenv global`, right? That seems not quite right. Still better than what we do now.

File: README.md (hunk 1 above)
Comment (novemberborn) [reply README.md]: It's been a while. Perhaps I meant to say `eval "$(nodenv init -)"`. IIRC I have a global Node.js version picked through `nodenv`, which was interfering with the tests.

File: bin/nodenv-package-json-engine (hunk 2 above)
Comment (novemberborn) [reply bin/nodenv-package-json-engine]: > I can't explicitly set nodenv shell to the same value as nodenv global, right?

You could, but I don't think that's relevant.

Comment (hurrymaplelad) [reply bin/nodenv-package-json-engine]: I mean this check will behave unexpectedly If `nodenv shell` is set to the same value as `nodenv global`. This test fails, using `package.json` version when I'd expect it to use `nodenv shell`:

```
@test 'Prefers `nodenv shell` over package.json' {
  create_version 4.2.1
  create_version 5.0.0
  cd_into_package 4.2.1
  nodenv global 5.0.0
  eval "$(nodenv sh-shell 5.0.0)"
  run nodenv package-json-engine
  assert_success ''
}
```

Seems like we need another way to detect if we're using the `nodenv global` value.

File: README.md (hunk 1 above)
Comment (hurrymaplelad) [reply README.md]: Let's leave this change out until we have a reproducible error. I suspect it's something we could fix without requiring contributors to jump through hoops. Perhaps `unset NODENV_VERSION` in test setup?

File: bin/nodenv-package-json-engine (hunk 2 above)
Comment (novemberborn) [reply bin/nodenv-package-json-engine]: > Seems like we need another way to detect if we're using the nodenv global value.

Perhaps always set a `NODENV_GLOBAL_VERSION` variable (if `nodenv global` is used)?

**Comment by hurrymaplelad** [comment]
Cannibalized with #29. Thanks for the PR!

**Pull request closed without merging**